### DISCUSSION TITLE
Help: Directory name and file name are merged together

### DISCUSSION THREAD

**Discussion by scottamain** [facebook/docusaurus#9513] [Q&A]
Hi folks, It appears that if a file name matches its folder name, Docusaurus collapses them (ignoring the filename), and I don't want that. 

For example, I have a file at `docs/libs/math/math.md`, but instead of rendering it where it should be at `libs/math/math/`, Docusaurus creates `libs/math/`, so any links to the correct path at `libs/math/math/` will 404. 

I know this redundant path name sounds silly, but it's the result of some generated docs and I can't reasonably change these names. And adding front matter for `slug` is a very ugly solution, because there should be nothing wrong with this file path.

I currently believe this could be a bug, because I can't find this behavior documented as part of the [Docusaurus routing behavior](https://docusaurus.io/docs/advanced/routing#routes-become-html-files). The only path/routing changes that are documented are ones like this:

> The `/docs/advanced/routing` route can correspond to either `/docs/advanced/routing/index.html` or `/docs/advanced/routing.html`

My experience indicates another possibility: `/docs/advanced/routing` might also correspond to `/docs/advanced/routing/routing.html`. That seems wrong to me.

Is there any way I can stop this? Thank you!

**Comment by Josh-Cena** [comment]
This is an intended feature, as part of the [category index convention](https://docusaurus.io/docs/sidebar/autogenerated#category-index-convention). And while it's possible to opt out from turning duplicate names into category indices, we intentionally does not allow the same configuration to apply to slug generation. You are right that there's no way to opt out, apart from using explicit `slug`. In a future minor version, we will introduce an option called `createFrontMatter` so that you can create the slug en masse.

**Reply by scottamain** [reply]
Thanks @Josh-Cena. So if this categorization of `lib/math/math.md` into `lib/math/` is intended behavior, would you agree it's a bug that `lib/math/math.html` results in 404?

**Reply by Josh-Cena** [reply]
> would you agree it's a bug that `lib/math/math.html` results in 404?

I'm not sure what you mean. `lib/math/math.md` generates one and only one URL, and that's `/lib/math`. Visiting any other URL, including `/lib/math/math.html`, is surely a 404.
